core/Updates/5.2.0-b2.php

```diff
@@ -79,20 +80,11 @@ private function detectCollationForMigration(): ?string
                 return $userTableCollation;
             }
 
-            $archiveTables = ArchiveTableCreator::getTablesArchivesInstalled(ArchiveTableCreator::NUMERIC_TABLE);
-
-            if (0 === count($archiveTables)) {
-                // skip if there is no archive table (yet)
-                return null;
-            }
-
-            // sort tables so we have them in order of their date
-            rsort($archiveTables);
-
-            $archiveTableStatus = $db->fetchRow('SHOW TABLE STATUS WHERE Name = ?', [$archiveTables[0]]);
+            $archiveTable = ArchiveTableCreator::getLatestArchiveTableInstalled(ArchiveTableCreator::NUMERIC_TABLE);
+            $archiveTableStatus = $metadataProvider->getTableStatus(Common::unprefixTable($archiveTable));
```

Comment:
I guess it's better to add a null check here again, as passing `null` to `Common::unprefixTable($archiveTable)` could result in a (deprecation) notice.

```suggestion
            if (null === $archiveTable) {
                return null;
            }

            $archiveTableStatus = $metadataProvider->getTableStatus(Common::unprefixTable($archiveTable));
```